cmd/topicmappr/commands/reassignments.go

```diff
@@ -288,6 +288,55 @@ func computeReassignmentBundles(
 	return results
 }
 
+/**
+getPartitionMapChunk Breaks a reassignment into a series of sequential, smaller reassignments.
+For large reassignments that may take a while, or risky operations that may require downtime in between, a chunked reassignment can be used.
+This will generate a series of partition maps that will converge on the desired state. To minimize data transfer,
+partitions are only moved to replicas in the desired final state.
+
+The original design was intended for downscaling operations, to remove partitions from one (or three) brokers at a time,
+without overwhelming whatever brokers are remaining in the cluster.
+*/
+func getPartitionMapChunk(finalMap *kafkazk.PartitionMap, initialMap *kafkazk.PartitionMap, brokerIds kafkazk.BrokerList, chunkStepSize int) []*kafkazk.PartitionMap {
+	var intermediateMap = initialMap.Copy()
+	var out []*kafkazk.PartitionMap
+	brokerIds.SortByIDDesc()
+
+	for i := 0; i < len(brokerIds); i += chunkStepSize {
+		// Select the brokers we will move data from for this chunk
+		var chunkBrokers = map[int]struct{}{}
+		for j := 0; j < chunkStepSize && i+j < len(brokerIds); j++ {
+			chunkBrokers[brokerIds[i+j].ID] = struct{}{}
+		}
+
+		// Go through the current map, and any partitions with replicas in our chunked brokers for this iteration
+		// will be moved this time.
+		var tempMap = intermediateMap.Copy()
+		for pIndex, p := range intermediateMap.Partitions {
+			for rIndex, r := range p.Replicas {
+				if _, correctReplica := chunkBrokers[r]; correctReplica {
+					// This replica needs to be switched with one from the final map
+					if len(tempMap.Partitions[pIndex].Replicas) != len(finalMap.Partitions[pIndex].Replicas) {
+						fmt.Println("Chunked reassignment cannot be used when reducing or increasing replication factor. Exiting.")
```

Comment:
Could this be validated early in the flags validation step (there is one, right? I'm going by memory)? it may help pull some of the nesting out here if so.

Reply:
Unfortunately I'm not sure it can, we only know if replication factor is being reduced inadvertently if theres not enough brokers, and the final output map has fewer. This check is good to have somewhere for correctness, but I'm not sure we really need it since you can only hit this scenario when running with ignore warnings.